### PR TITLE
Added mail generator and verification function.

### PR DESCRIPTION
**Pull request by yawn**
## Definition of done (v0.0.1)

- [x] Automated integration test - will come later for the whole feature (see also [notes](https://github.com/superwerker/superwerker/issues/13#issuecomment-708531741) from #13)
- [X] Integrity protection - not needed
- [X] Structured Logging and Monitoring for Lambda - this is actually a proposal for some really simplistic logging; monitoring could be spiked here with X-Ray; wdyt?

The plan would be to reference the lambda output in the parent stack and either override the email (e.g. for existing CT setups in the CT stack) or let the function generate them. By default it also verifies (while doing exponential backoffs) that the email delegation setup works.

As a followup I'd delete the obsolete healthcheck.

### CODE REVIEW

**Comment by s0enke** [comment]
Could  d6f83d5 go into another PR so this PR would be more contained and easier to review/merge?

**Comment by yawn** [comment]
Good to go from my POV

File: tests/rootemails.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    @classmethod
+    def send_email(cls, id, body):
+
+        res = ses.list_identities(
+            IdentityType='Domain',
+            MaxItems=1,
+        )
+
+        domain = res['Identities'][0]
```
Comment (yawn) [inline tests/rootemails.py:21]: We probably should persist the identity as SSM parameter from within the stack.

File: components/rootmail.yaml [hunk 2]
```diff
@@ -69,6 +72,131 @@ Resources:
...
+          time.sleep(n)
+
+        def handler(event, context):
+
+          log({
+            'event': event,
+            'level': 'debug',
+          })
+
+          verify = event.get('verify', 'true')
```
Comment (s0enke) [inline components/rootmail.yaml:110]: Not sure if mixing generator and email setup verification into one function is good? I'd rather have two functions and separation of concerns.

File: tests/rootemails.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                    'Text': {
+                        'Data': body,
+                    },
+                },
+            },
+        )
+
+        return res
+
+    def test_root_email(self):
```
Comment (s0enke) [inline tests/rootemails.py:44]: Nice test!

File: tests/rootemails.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+        id = res['OpsItemSummaries'][0]['OpsItemId']
+
+        res = ssm.get_ops_item(
+            OpsItemId=id,
+        )
+
+        self.assertEqual('This is a mail body', res['OpsItem']['Description'].rstrip())
+
+        ssm.update_ops_item(
```
Comment (s0enke) [inline tests/rootemails.py:81]: We could also remove all open ops items in the cleanup stage of the pipeline

File: tests/rootemails.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+        self.assertEqual(1, len(res.get('OpsItemSummaries', [])))
+
+        id = res['OpsItemSummaries'][0]['OpsItemId']
+
+        res = ssm.get_ops_item(
+            OpsItemId=id,
+        )
+
+        self.assertEqual('This is a mail body', res['OpsItem']['Description'].rstrip())
```
Comment (s0enke) [inline tests/rootemails.py:79]: Nitpick: This could be variable like `id` as well.

File: components/rootmail.yaml [hunk 6]
```diff
@@ -370,54 +498,91 @@ Resources:
...
+
+              ssm.create_ops_item(
+                Description=description,
+                OperationalData=operational_data,
+                Source=source,
+                Title=title,
+              )
+
+        def log(msg):
+          print(json.dumps(msg), flush=True)
```
Comment (s0enke) [inline components/rootmail.yaml:588]: Python also has a native `logging` lib, which could be used. But this simple json dump is OK for me for now.

**Review by s0enke — APPROVED** [review]

**Pull request merged**